gen/list.go

```diff
@@ -60,7 +60,7 @@ func (l *listGenerator) ValueList(g Generator, spec *compile.ListSpec) (string,
 				<- else ->
 				for <$i>, <$x> := range <$v> {
 					if <$x> == nil {
-						return <import "fmt">.Errorf("invalid [%v]: value is nil", <$i>)
+						return <import "fmt">.Errorf("invalid '<typeReference .Spec>', index [%v]: value is nil", <$i>)
```

Comment:
nit: map and set error messages say "invalid map '..'" or "invalid set '...'" but this says just "invalid '...'".

I don't have a strong preference between "invalid list '...'" for this, or "invalid '...'" for all three, so your call on which error message you like more.

Reply:
Made the error message more explicit with "invalid list '...'".